File: source/backend/common/Validation.ts
/**
 * 入力値の妥当性確認
 */

import { SendMoneyCommand } from "../application/port/in/SendMoneyCommand";

export class Validation {
    static sendMoneyCommand(input: SendMoneyCommand) {
        if (input.sourceAccount.getAccountId().length > 0) return false;
        if (input.targetAccount.getAccountId().length > 0) return false;
        if (typeof input.money.amount === "number" && input.money.amount > 0) return false;

        // 本来はここでエラーを投げる
        return true;
    }
}

File: source/backend/python/common/validation.py
from application.port.ingress.send_money_command import SendMoneyCommand

class Validation:
    @staticmethod
    def send_money_command(input_params: SendMoneyCommand):
        if len(input_params.source_account.get_account_id()) > 0:
            return False
        if len(input_params.target_account.get_account_id()) > 0:
            return False
        if type(input_params.money.amount) is int and input_params.money.amount > 0:
            return  False

        return True